### PR TITLE
WIP: Adding wordcloud 1.2.1

### PR DESCRIPTION
**Pull request by ericdill**

### CODE REVIEW

**Comment by ocefpaf** [comment]
@ericdill for some reason GitHub thinks that the `meta.yaml` file is a binary. Can you check what is going on?

**Comment by ericdill** [comment]
@ocefpaf Yeah I'm trying to do this from my windows machine that I never actually write code on, so I'm sure it's something dumb that I've done.  Working on converting this file.... 

:facepunch: windows...

**Comment by ericdill** [comment]
Huh. That's interesting... I have the `sha256` key with no value and *conda build doesn't fail with a sha256 mismatch error*.  That does not seem like the right behavior.  attn @msarahan

**Comment by ocefpaf** [comment]
> Huh. That's interesting... I have the `sha256` key with no value and *conda build doesn't fail with a sha256 mismatch error*.  That does not seem like the right behavior.  attn @msarahan

:scream:

File: recipes/wordcloud/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set version = "1.2.1" %}
+
+package:
+    name: wordcloud
+    version: {{ version }}
+
+source:
+    url: https://github.com/amueller/word_cloud/archive/{{ version }}.tar.gz
+    fn: wordcloud-{{ version }}.tar.gz
+    sha256:
```
Comment (jakirkham) [inline recipes/wordcloud/meta.yaml]: So are we going to fill this guy in or were we waiting for `conda-build` to start failing? 😜

File: recipes/wordcloud/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        - nosetests {{ environ.SRC_DIR }}/test -sv
+
+about:
+    home: https://github.com/amueller/word_cloud
+    license: MIT
+    summary: A little word cloud generator in Python
+
+extra:
+    recipe-maintainers:
+        - ericdill
```
Comment (jakirkham) [inline recipes/wordcloud/meta.yaml]: @amueller, any interest in helping maintain `wordcloud`?

File: recipes/wordcloud/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    sha256:
+
+build:
+    number: 0
+    script: python setup.py install
+    entry_points:
+        - wordcloud = wordcloud.wordcloud_cli
+
+requirements:
+    build:
```
Comment (jakirkham) [inline recipes/wordcloud/meta.yaml]: Could we please add `toolchain`?

File: recipes/wordcloud/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    version: {{ version }}
+
+source:
+    url: https://github.com/amueller/word_cloud/archive/{{ version }}.tar.gz
+    fn: wordcloud-{{ version }}.tar.gz
+    sha256:
+
+build:
+    number: 0
+    script: python setup.py install
```
Comment (jakirkham) [inline recipes/wordcloud/meta.yaml]: Please change to

```
    script:
        # FIXME: This is a hack to make sure the environment is activated.
        # The reason this is required is due to the conda-build issue
        # mentioned below.
        #
        # https://github.com/conda/conda-build/issues/910
        #
        - source activate "${CONDA_DEFAULT_ENV}"  # [unix]
        - python setup.py install
```

**Comment by amueller** [comment]
I'd be very happy to help. I wanted to do this for a while. I know very little about the conda process right now but I need to get up to speed for scikit-learn anyhow.

File: recipes/wordcloud/meta.yaml (hunk 1 above)
Comment (ericdill) [inline recipes/wordcloud/meta.yaml]: Yeah fair enough. I'll fill it in

**Comment by ericdill** [comment]
Thanks for wanting to help @amueller !   And thanks for pinging him @jakirkham !

Regarding entries in `build/entry_points`... Do they have to point at a function or can they target the `if __name__ == "__main__":` block too?  In this wordcloud package, the cli entry point runs through such a `if __name__ == "__main__":` block, [here] (https://github.com/amueller/word_cloud/blob/1.2.1/wordcloud/wordcloud_cli.py#L81).  

I'm not having any success with trying to hit that block through conda's entry_points with this code:
```yaml
    entry_points:
        - wordcloud = wordcloud.wordcloud_cli
```

I *could* patch in a simple run() function that does the same thing as the `if __name__ == "__main__":` bit, but that feels wrong for some reason.  Though if that is the recommended method, I'm perfectly happy to do that.

It's pretty simple to patch in something like this:

```diff
diff --git a/wordcloud/wordcloud_cli.py b/wordcloud/wordcloud_cli.py
index 2a54deb..5628ca8 100644
--- a/wordcloud/wordcloud_cli.py
+++ b/wordcloud/wordcloud_cli.py
@@ -77,5 +77,10 @@ def parse_args(arguments):
     args.color_func = color_func
     return args
 
-if __name__ == '__main__':
+
+def run():
     main(parse_args(sys.argv[1:]))
+
+
+if __name__ == "__main__":
+    run()
```

and then have the entry_points look like this:

```yaml
    entry_points:
        - wordcloud = wordcloud.wordcloud_cli:run
```

and then conda build is happy.  Thoughts @jakirkham @ocefpaf ?

**Comment by amueller** [comment]
I'm happy with that patch going upstream if it makes the declaration of the entry point easier.

**Comment by ericdill** [comment]
Upon further investigation, looks like the entry_points section is not entirely necessary.  It seems that I can use the "wordcloud_cli.py" executable that gets created in setup.py, at least on linux.  Let's see if it actually works on the other platforms.  Is this bad practice to *not* define an entry_point in the conda package even though we know there is one defined in setup.py?

File: recipes/wordcloud/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+build:
+    number: 0
+    script:
+      # FIXME: This is a hack to make sure the environment is activated.
+      # The reason this is required is due to the conda-build issue
+      # mentioned below.
+      #
+      # https://github.com/conda/conda-build/issues/910
+      #
+      - source activate "${CONDA_DEFAULT_ENV}"  # [unix]
```
Comment (jakirkham) [inline recipes/wordcloud/meta.yaml]: We can drop this line and the comments above now. 😄

Comment (ericdill) [inline recipes/wordcloud/meta.yaml]: Hah alright.  I've been checked-out from conda-forge for the last couple of months. I've missed so much!  Is the current thinking back to `python setup.py install` for distutils and add `[[-single-version-externally-managed --record=record.txt]]` for setuptools?

Comment (jakirkham) [inline recipes/wordcloud/meta.yaml]: Yeah, that was a pretty recent change actually like 2-3 weeks ago.

Currently yes. Though there is some discussion on that last point.

**Comment by ericdill** [comment]
rebased and pushed to trigger CIs again.  The circleci failure seems intermittent. Hopefully it'll work now...

**Comment by ericdill** [comment]
This is ready to go. Thanks for the feedback/help/comments @amueller @jakirkham @ocefpaf

**Comment by ericdill** [comment]
thanks @ocefpaf !

**Pull request merged**